### ISSUE TITLE
Trying separable filters for better performance

### ISSUE THREAD

**Issue by tech4GT** [publiclab/image-sequencer#1059]
So, I had this idea that if we can modify pixel manipulation api to support separable filters, then we can speed up our performance for stuff like edge detection quite a bit.
Since that takes us from from O(n^2) to O(2*n) or simply O(n).
Or another fast way to speed up convolution can be fourier transform?
Just brainstorming here.

@jywarren thoughts

**Comment by tech4GT** [comment]
Also, I am aware that we are currently using pretty small filters so this wont make a huge difference but in case that we want to use larger filters in the future this might be a nice upgrade overall! :D

**Comment by HarshKhandeparkar** [comment]
@tech4GT yes! We have 5x5 Gaussian filter which is separabale. Everything is GPU accelerated. We olnly need to change `src/modules/_nomodule/gpuUtils.js` to support separable filters.

**Comment by tech4GT** [comment]
Great! I suppose we can even generalize the gaussian! This might be a good idea for summer.

**Comment by HarshKhandeparkar** [comment]
We should also work on 5x5 general purpose convolution module.
